### PR TITLE
integration: simplify parallel run destination

### PR DESCRIPTION
**Pull request by thaJeztah**
splitting this one from https://github.com/moby/moby/pull/39724 as well, hopefully to make the other PR easier to review

'Namespace' parallel runs by bind-mounting a different directory
in the container, instead of making the tests running inside
the container aware of the namespaced location.

This makes it transparent to the tests, and slightly reduces
complexity.

This change fixes the problem I ran into in https://github.com/moby/moby/pull/39724#issuecomment-524450523:

-----

Hm; looks like there's still an issue (at least with the uploaded bundles); with debug enabled, I see that these tests are run. Which means that we should have in total **18 go-test-report.json** files, and **18 junit-report.xml** files.

```
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-build-go-test-report.json              --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-build-junit-report.xml              --raw-command -- go tool test2json -p amd64.integration.build              -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-config-go-test-report.json             --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-config-junit-report.xml             --raw-command -- go tool test2json -p amd64.integration.config             -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-container-go-test-report.json          --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-container-junit-report.xml          --raw-command -- go tool test2json -p amd64.integration.container          -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-image-go-test-report.json              --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-image-junit-report.xml              --raw-command -- go tool test2json -p amd64.integration.image              -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-go-test-report.json            --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-junit-report.xml            --raw-command -- go tool test2json -p amd64.integration.network            -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-ipvlan-go-test-report.json     --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-ipvlan-junit-report.xml     --raw-command -- go tool test2json -p amd64.integration.network.ipvlan     -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-macvlan-go-test-report.json    --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-network-macvlan-junit-report.xml    --raw-command -- go tool test2json -p amd64.integration.network.macvlan    -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-go-test-report.json             --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-junit-report.xml             --raw-command -- go tool test2json -p amd64.integration.plugin             -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-authz-go-test-report.json       --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-authz-junit-report.xml       --raw-command -- go tool test2json -p amd64.integration.plugin.authz       -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-common-go-test-report.json      --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-common-junit-report.xml      --raw-command -- go tool test2json -p amd64.integration.plugin.common      -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-graphdriver-go-test-report.json --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-graphdriver-junit-report.xml --raw-command -- go tool test2json -p amd64.integration.plugin.graphdriver -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-logging-go-test-report.json     --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-logging-junit-report.xml     --raw-command -- go tool test2json -p amd64.integration.plugin.logging     -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-volumes-go-test-report.json     --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-plugin-volumes-junit-report.xml     --raw-command -- go tool test2json -p amd64.integration.plugin.volumes     -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-secret-go-test-report.json             --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-secret-junit-report.xml             --raw-command -- go tool test2json -p amd64.integration.secret             -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-service-go-test-report.json            --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-service-junit-report.xml            --raw-command -- go tool test2json -p amd64.integration.service            -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-session-go-test-report.json            --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-session-junit-report.xml            --raw-command -- go tool test2json -p amd64.integration.session            -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-system-go-test-report.json             --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-system-junit-report.xml             --raw-command -- go tool test2json -p amd64.integration.system             -t ./test.main -test.v -test.timeout=120m
test_env gotestsum --format=standard-verbose --jsonfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-volume-go-test-report.json             --junitfile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-integration-volume-junit-report.xml             --raw-command -- go tool test2json -p amd64.integration.volume             -t ./test.main -test.v -test.timeout=120m
```

However, in the "janky" bundle, I only see; **8 JSON** files, and **6 XML** files, so either the `find` is incorrect, or somewhere we purge files, or files are stored somewhere nested and therefore not found?

```
amd64-integration-config-go-test-report.json
amd64-integration-container-go-test-report.json
amd64-integration-network-ipvlan-go-test-report.json
amd64-integration-network-macvlan-go-test-report.json
amd64-integration-plugin-authz-go-test-report.json
amd64-integration-plugin-go-test-report.json
amd64-integration-plugin-logging-go-test-report.json
amd64-integration-service-go-test-report.json
```

```
amd64-integration-build-junit-report.xml
amd64-integration-plugin-graphdriver-junit-report.xml
amd64-integration-plugin-logging-junit-report.xml
amd64-integration-plugin-volumes-junit-report.xml
amd64-integration-service-junit-report.xml
amd64-integration-session-junit-report.xml
```

All files _are_ present in the PowerPC and s390x bundles, so I suspect that it's either because the path is different for the Janky run (they use the `1/`, `2/, `3/` subdirectories, or perhaps there's an issue because of the test-flags (it's weird though, because for some tests the _XML_ is there, but the _JSON_ is missing, and vice-versa)

### CODE REVIEW

**Comment by thaJeztah** [comment]
@kolyshkin @tiborvass PTAL

File: hack/make/test-integration [hunk 1]
```diff
@@ -1,12 +1,6 @@
 #!/usr/bin/env bash
 set -e -o pipefail
 
-if [ -n "$TEST_INTEGRATION_DEST" ]; then
-	export DEST="$ABS_DEST/$TEST_INTEGRATION_DEST"
-	export DOCKER_INTEGRATION_DAEMON_DEST="$DEST"
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply hack/make/test-integration]: it wasn't there before https://github.com/moby/moby/commit/e554fb23c82bd672436800dac3ea9739da8e3377, so looked like I could remove it to be back to what was there before

[the remark this answers is not in the document]
Comment (thaJeztah) [reply hack/make/test-integration]: Wondering if the problem I ran into is somehow caused by https://github.com/moby/moby/commit/13df617d4cc98fe31f3ded78902119a1c1505005

https://github.com/moby/moby/blob/13df617d4cc98fe31f3ded78902119a1c1505005/hack/make/.integration-daemon-stop#L13-L14

[the remark this answers is not in the document]
Comment (thaJeztah) [reply hack/make/test-integration]: hm, or multiple containers mounting the same path, could unmounting (when the container is stopped) cause issues for another container mounting the same path?

**Comment by andrewhsu** [comment]
I talked with @thaJeztah about this and I think this should help to simplify the isolation. cc @tiborvass

**Review by andrewhsu — APPROVED** [review]
LGTM on green

**Review by seemethere — APPROVED** [review]

**Review by tiborvass — APPROVED** [review]

**Pull request merged**